src/hotspot/cpu/aarch64/templateInterpreterGenerator_aarch64.cpp

```diff
@@ -1272,6 +1272,9 @@ address TemplateInterpreterGenerator::generate_native_entry(bool synchronized) {
   }
 #endif
 
+  // AOT training run support
+  __ end_training_check();
```

Comment:
Does it make sense to do these checks after `notify_method_entry()`? It could aid in debugging/testing that the training run indeed ended on this method entry.

Reply:
good call, will swap order of notify_method_entry and end_training_check